spec/integration/manifest.spec.yml

```diff
@@ -309,6 +309,10 @@ spec:
       $ref: "./data_stream/manifest.spec.yml#/definitions/vars"
     owner:
       $ref: "#/definitions/owner"
+    allow_snapshot:
+      type: boolean
+      description: Can assets be exported from a SNAPSHOT version?
+      default: false
```

Comment:
In https://github.com/elastic/package-spec/issues/318#issuecomment-1096811751, it is mentioned that if it exists a label, package could contain assets built from Elastic stack SNAPSHOT versions.

My first approach to achieve this is adding a new field `allow_snapshot` into the spec of the root package. This field must be added explicitly in case of stable package versions if they need to use assets built from Elastic stack SNAPSHOT versions. WDYT ? Any other suggestion for this ? @elastic/ecosystem

Reply:
I'm not convinced about the approach as the manifest describes the package content, but `allow_snapshot` is related to the procedure of building/exporting assets.

> These warnings will enforce to create/export Kibana assets using stable Elastic stack versions

Shouldn't we warn/block the `export` command if this isn't a stable stack? We could detect the Kibana/ES version instead. I just feel that we don't really don't need a new manifest field :)

Reply:
Agreed, it does not feel the place to add a field for that purpose. This first approach was to try to develop everything under the package-spec umbrella.

I'll check the export command you suggested for options.
Maybe it can be done something like:
- in package-spec, the validation rule shows warning messages if assets are built from Elastic Stack
    - Exceptions: technical preview (0.x.x) and prerelease packages that could have those kind of assets.
- in elastic-package:
    - export dashboard fails or shows a warning if it is from an SNAPSHOT version
    - add a parameter to export command `-allow-snapshot`, so developers need to add it explicitly if they need those dashboards:
      ```
      elastic-package export dashboards -allow-snapshot
      ``` 
        - warning messages defined by package-spec would be shown.

WDYT about this other approach?

Reply:
I mean, from the package-spec perspective it shouldn't be bad to use snapshots, so we shouldn't warn about it? I would see this feature rather scoped within the `export` command with a switch (as you suggested), but wouldn't leak the logic to spec.

Feel free to discuss the idea also with Jaime as I might be biased here (used to work with spec since it was born.)

Reply:
Yeah, I agree that the versions used in the dashboards are not so related to a package specification, it is not that these versions are _wrong_.

I like the option of making `elastic-package export dashboards` to fail unless `-allow-snapshot` is used. This is a way to make the developer decide if they know what they are doing. We can also start with this option, and add more warnings in the future if this is not enough.

Another option could be to add the warning in `elastic-package`, as `elastic-package` can be more opinionated towards best-practices, while the spec is more focused on producing valid packages.

But summarizing, I think we can start by adding the `-allow-snapshot` flag to `elastic-package export dashboards`, and make it fail, with a clear error message, if snapshots are found when this flag is not used.

Reply:
Btw, food for thought, the main concern with the use of dashboards generated by snapshot versions is that the developer may be using features that are not released at the end, or are in a different way. If a dashboard breaks in a stable version because of this, it could be detected by tests on dashboards, that would be nice to have at some point.

Reply:
Ok! It's true that dashboards are not wrong even if they are being exported from a SNAPSHOT version in the scope of the spec. As you both suggested, this PR can be closed in favor of https://github.com/elastic/elastic-package/pull/986 where I'm adding that logic in the export command.

Thanks @mtojek and @jsoriano for your comments here!!

Reply:
> If a dashboard breaks in a stable version because of this, it could be detected by tests on dashboards, that would be nice to have at some point.

If CI could detect those scenarios, that would be even better